x-pack/plugins/ml/server/lib/ml_client/search.ts

```diff
@@ -34,11 +34,15 @@ export function searchProvider(
   ): Promise<TransportResult<estypes.SearchResponse<T>, unknown>> {
     await jobIdsCheck('anomaly-detector', jobIds);
     const { asInternalUser } = client;
-    const resp = await asInternalUser.search<T>({
-      ...searchParams,
-      index: ML_RESULTS_INDEX_PATTERN,
-    });
+    const resp = await asInternalUser.search<T>(
+      {
+        ...searchParams,
+        index: ML_RESULTS_INDEX_PATTERN,
+      },
+      { meta: true }
```

Comment:
`anomalySearch` is a simple wrapper around a standard es `search` call.
For consistency with how other `search` calls are made I think this should also not return a `body`

Reply:
Yea, except it's using it's own API definition instead of inheriting it from the `Client` interface. Plus, the actual definition is inferred from the provider function.

https://github.com/elastic/kibana/blob/17fcc359a4c3aa1a03e04c93e10add21f768fbe6/x-pack/plugins/ml/server/lib/ml_client/types.ts#L13-L15

For consistency, it would require  a proper definition with the 3 overrides, as done on every client API def

e.g

```ts
closeJob(this: That, params: T.MlCloseJobRequest | TB.MlCloseJobRequest, options?: TransportRequestOptionsWithOutMeta): Promise<T.MlCloseJobResponse>;
closeJob(this: That, params: T.MlCloseJobRequest | TB.MlCloseJobRequest, options?: TransportRequestOptionsWithMeta): Promise<TransportResult<T.MlCloseJobResponse, unknown>>;
closeJob(this: That, params: T.MlCloseJobRequest | TB.MlCloseJobRequest, options?: TransportRequestOptions): Promise<T.MlCloseJobResponse>;
 ```

Plus adapt the provider (`x-pack/plugins/ml/server/lib/ml_client/search.ts`) to handle the `meta` parameter as I did on the other non-passthough functions of `MlClient`. (and then of course adapt all usages)

Do you think it would be acceptable to have this considered as cleanup and be performed by the actual owners as follow-up?

Reply:
nm, done in https://github.com/elastic/kibana/pull/124488/commits/1ff4d928db3cf3319947a4aa2203a5b59a27f952

Reply:
brilliant, thank you.